aws/cloudfront/cloudfront.rb

```diff
@@ -0,0 +1,150 @@
+require 'cdo/session'
+
+dashboard_hostname = CDO.canonical_hostname('studio.code.org')
+pegasus_hostname = CDO.canonical_hostname('code.org')
+
+# Cloudfront-specific configuration.
+CDO.cloudfront = {
+  pegasus: {
+    aliases: [pegasus_hostname],
+    origin: "#{ENV['RACK_ENV']}-pegasus.code.org",
+    # IAM server certificate name
+    ssl_cert: 'codeorg-cloudfront',
+    log: {
+      bucket: 'cdo-logs',
+      prefix: "#{ENV['RACK_ENV']}-pegasus-cdn"
+    }
+  },
+  dashboard: {
+    aliases: [dashboard_hostname],
+    origin: "#{ENV['RACK_ENV']}-dashboard.code.org",
+    ssl_cert: 'codeorg-cloudfront',
+    log: {
+      bucket: 'cdo-logs',
+      prefix: "#{ENV['RACK_ENV']}-dashboard-cdn"
+    }
+  }
+}
+
+require 'cookbooks/cdo-varnish/libraries/http_cache'
+CDO.http_cache = HttpCache.config(Session::KEY, Session::STORAGE_ID)
+
+# Returns a CloudFront DistributionConfig Hash compatible with the AWS SDK for Ruby v2.
+# Syntax reference: http://docs.aws.amazon.com/sdkforruby/api/Aws/CloudFront/Types/DistributionConfig.html
+def cloudfront_config(cloudfront, config, reference = nil)
+  behaviors = config[:behaviors].map do |behavior|
+    if behavior[:path].is_a? Array
+      behavior[:path].map do |path|
+        cache_behavior behavior, path
+      end
+    else
+      cache_behavior behavior, behavior[:path]
+    end
+  end.flatten
+
+  ssl_cert = cloudfront[:ssl_cert] &&
+  {
+    aliases: {
+      quantity: cloudfront[:aliases].length, # required
+      items: cloudfront[:aliases].empty? ? nil : cloudfront[:aliases],
+    },
+    default_root_object: '',
+    origins: {# required
+      quantity: 1, # required
+      items: [
+        {
+          id: 'cdo', # required
+          domain_name: cloudfront[:origin], # required
+          origin_path: '',
+          custom_origin_config: {
+            http_port: 80, # required
+            https_port: 443, # required
+            origin_protocol_policy: 'match-viewer', # required, accepts http-only, match-viewer
+          },
+        },
+      ],
+    },
+    default_cache_behavior: cache_behavior(config[:default]),
+    cache_behaviors: {
+      quantity: behaviors.length, # required
+      items: behaviors.empty? ? nil : behaviors,
+    },
+    custom_error_responses: {
+      quantity: 0 # required
+    },
+    comment: '', # required
+    logging: {
+      enabled: !!cloudfront[:log], # required
+      include_cookies: false, # required
+      bucket: cloudfront[:log] && "#{cloudfront[:log][:bucket]}.s3.amazonaws.com", # required
+      prefix: cloudfront[:log] && cloudfront[:log][:prefix], # required
+    },
+    price_class: 'PriceClass_All', # accepts PriceClass_100, PriceClass_200, PriceClass_All
+    enabled: true, # required
+    viewer_certificate: cloudfront[:ssl_cert] ? {
+      # Lookup IAM Certificate ID from server certificate name
+      iam_certificate_id: Aws::IAM::Client.new
+        .get_server_certificate(server_certificate_name: cloudfront[:ssl_cert])
```

Comment:
Style/DotPosition: Place the . on the previous line, together with the method call receiver.